flexmeasures/data/models/data_sources.py

```diff
@@ -90,10 +90,11 @@ def __str__(self) -> str:
     def to_dict(self) -> dict:
         model_incl_version = self.model if self.model else ""
         if self.model and self.version:
-            model_incl_version += f" ({self.version}"
+            model_incl_version += f" (v{self.version})"
         return dict(
             id=self.id,
             name=self.name,
             model=model_incl_version,
+            type=self.type if self.type in ("forecaster", "scheduler") else "other",
```

Comment:
I don't see why we are restricting to express various source types at this point. Is some code we wrote elsewhere dependent on the type being only one of three strings?

It seems a bit at odds with the new CLI command which allows all kind of types.

Reply:
> Is some code we wrote elsewhere dependent on the type being only one of three strings?

Yes. They are used to distinguish **stroke dash** in charts. Basically, data from any source will be plotted as a solid line, except data from forecasters (dotted) and schedulers (dashed). See `create_line_layer` in `data/models/charts/belief_charts.py`. I'll add a note or cross-reference here, as well.

I fought valiantly, but ultimately failed to make the stroke dash domain adjust automatically to the source types in the shown data. That means that the legend now always shows these three source types, regardless of what data is shown. I decided that this isn't so bad for our purpose. And a side benefit is that this allows us to precisely control the vertical order of the legend items.

Reply:
But this code (`to_dict()`) is not really just connected to charts. Data sources van have any type, but we are not including them here if they are not one of the three we support in charts.
This exclusion should happen in code that deals with charting, right?

Reply:
Ideally, yes. I'll give it another shot.